packages/compiler/src/experimental/mutators.ts

```diff
@@ -132,10 +250,19 @@ const seen = new CustomKeyMap<[MutableType, Set<Mutator> | Mutator[]], Type>(([t
 });
 
 /**
- * Mutate the type graph with some namespace mutation.
- * **Warning** this will most likely end up mutating the entire TypeGraph
- * as every type relate to namespace in some way or another
- * causing parent navigation which in turn would mutate everything in that namespace.
+ * Mutate the type graph, allowing namespaces to be mutated.
+ *
+ * **Warning**: This function will likely mutate the entire type graph. Most TypeSpec types relate to namespaces
```

Comment:
Might be good to add two things here: it will only mutate the entire type graph if you mutate namespaces, and if you don't want to mutate namespaces then you should use @ link mutateSubgraph.